python/pyiceberg/catalog/hive.py

```diff
@@ -0,0 +1,262 @@
+        except NoSuchObjectException as e:
+            raise NoSuchTableError(f"Table {from_database_name}.{from_table_name} not found") from e
+        except InvalidOperationException as e:
+            raise NoSuchNamespaceError(f"Destination database {to_database_name} does not exists") from e
+        return Table()
+
+    def create_namespace(self, namespace: Union[str, Identifier], properties: Optional[Properties] = None) -> None:
+        database_name = self.identifier_to_tuple(namespace)[0]
+
+        hive_database = HiveDatabase(name=database_name, parameters=properties)
+
+        try:
+            with self._client as open_client:
+                open_client.create_database(hive_database)
+        except AlreadyExistsException as e:
+            raise AlreadyExistsError(f"Database {database_name} already exists") from e
+
+    def drop_namespace(self, namespace: Union[str, Identifier]) -> None:
+        database_name = self.identifier_to_tuple(namespace)[0]
+        try:
+            with self._client as open_client:
+                open_client.drop_database(database_name, deleteData=True, cascade=False)
+        except InvalidOperationException as e:
+            raise NamespaceNotEmptyError(f"Database {database_name} is not empty") from e
+        except MetaException as e:
+            raise NoSuchNamespaceError(f"Database {database_name} does not exists") from e
+
+    def list_tables(self, namespace: Optional[Union[str, Identifier]] = None) -> List[Identifier]:
+        database_name = self.identifier_to_tuple(namespace)[0] if namespace else namespace
+        with self._client as open_client:
+            return open_client.get_all_tables(db_name=database_name)
+
+    def list_namespaces(self) -> List[Identifier]:
+        with self._client as open_client:
+            return open_client.get_all_databases()
+
+    def load_namespace_properties(self, namespace: Union[str, Identifier]) -> Properties:
+        database_name = self.identifier_to_tuple(namespace)[0]
+        try:
+            with self._client as open_client:
+                database = open_client.get_database(database_name)
+                return database.parameters
+        except NoSuchObjectException as e:
+            raise NoSuchNamespaceError(f"Database {database_name} does not exists") from e
+
+    def update_namespace_properties(
+        self, namespace: Union[str, Identifier], removals: Optional[Set[str]] = None, updates: Optional[Properties] = None
+    ) -> None:
+        database_name = self.identifier_to_tuple(namespace)[0]
+        with self._client as open_client:
+            database = open_client.get_database(database_name)
+            if updates:
+                database.parameters.update(updates)
+            if removals:
+                for key in removals:
+                    if key in database.parameters:
+                        del database.parameters[key]
```

Comment:
The Java client specifically sets these to `null`. I think the purpose is to tell the HMS service to drop the keys.

Reply:
Great catch, thanks!